amd64/root/start.sh

```diff
@@ -54,7 +54,19 @@ if [ "$DECONZ_VNC_MODE" != 0 ]; then
     fi
 
     # Generate SSL certificate
-    openssl req -x509 -nodes -newkey rsa:2048 -keyout /root/.vnc/novnc.pem -out /root/.vnc/novnc.pem -days 365 -subj "/CN=deconz"
+    if [ -f "/root/.vnc/novnc.pem" ]; then
+      ENDDATE=`openssl x509 -noout -in /root/.vnc/novnc.pem -enddate | sed -e 's#notAfter=##'`
+      END=`date -d "${ENDDATE}" '+%s'`
+      NOW=`date '+%s'`
+      DIFF="$((${END}-${NOW}))"
+      if [ "${DIFF}" -lt 0 ]; then
+        echo "[marthoc/deconz] The certificate has expired!"
+      else
+        echo "[marthoc/deconz] The certificate will expire in $((${DIFF}/3600/24)) days."
```

Comment:
If we are going to check if the certificate is expired, then we should also give some direction to the user based on that info, and, in the current iteration of this feature, delete the old cert and generate a new one (otherwise, how is the user to fix this issue? We don't want the user to have to bash into the container to delete the expired cert). 

I think the log message about how long it will take the certificate to expire is unnecessary. But the log message telling the user that the cert is expired could say:
```[marthoc/deconz] The noVNC SSL certificate has expired; generating a new certificate now.```

We will need to adjust this behaviour if we ever allow users to use their own certificate.

I am open to your thoughts on this.

Reply:
Yes, this should be kept more simple - and this script has to fallback to generate a new certificate. It just shouldn't do this every time.

I also would put the filename (or at least the path) into a variable as this is repeated quit often:
```sh
    # Assert valid SSL certificate
    NOVNC_CERT="/root/.vnc/novnc.pem"
    if [ -f "$NOVNC_CERT" ] &&
        openssl x509 -noout -in "$NOVNC_CERT" -checkend 0 >/dev/null; then
        echo "The noVNC SSL certificate has expired; generating a new certificate now."
        rm "$NOVNC_CERT"
    fi
    if [ ! -f "$NOVNC_CERT" ]; then
      openssl req -x509 -nodes -newkey rsa:2048 -keyout "$NOVNC_CERT" -out "$NOVNC_CERT" -days 365 -subj "/CN=deconz"
    fi
```
`-checkend 0` would check if the certificate expires in zero seconds. On the other hand it won't be very useful to use an certificate which will expire in 10 seconds ... 1 minute ... 1 hour? So feel free to use a value with more sustainability.

Reply:
I will take your contribution for now.